lib/typhoeus/normalized_header_hash.rb

```diff
@@ -52,7 +52,7 @@ def delete(key)
 
   private
     def convert_key(key)
-      key.to_s.split(/_|-/).map { |segment| segment.capitalize }.join("-")
+      key.to_s.tr('_'.freeze,'-'.freeze).split('-'.freeze).map { |segment| segment.capitalize.freeze }.join('-'.freeze)
```

Comment:
I did some more benchmarks, performance-wise it seems to make no real difference between frozen strings and constants. So whatever looks better style-wise should be fine.